apps/accounts/views.py

```diff
@@ -89,7 +92,12 @@ def inner(self, request):
                     user = request.user
             else:
                 user = identity_user
-            return fn(self, request, user=user, identity=identity)
+            if len(state_parts) == 2:
+                next_path = base64.urlsafe_b64decode(str(state_parts[1]))
```

Comment:
`window.location` seems to encode unicode characters in some way that this seems to work. That being said there's nothing prevent someone from putting whatever they want in there and this blowing up. I'll some logging.